OpenRA.Mods.RA2/Traits/SpawnNeighboringActors.cs

```diff
@@ -0,0 +1,86 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	[Desc("This actor places other actors around itself.")]
+	public class SpawnNeighboringActorsInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[ActorReference]
+		[Desc("Types of actors to place. If multiple are defined, a random one will be selected.")]
+		public readonly HashSet<string> ActorTypes = new HashSet<string>();
+
+		[FieldLoader.Require]
+		[Desc("Locations to place the mines, from top-left of the building.")]
```

Comment:
```suggestion
		[Desc("Locations to spawn the actors, relative to the top-left of the building.")]
```

Reply:
Is `self.Location` _guaranteed_ to be the top-left (and if so, for my edification, how/where)?
If not, I think we should say something like "relative to the origin of this actor."

Reply:
Look at the `Building` code or place an actor in the editor. The origin is always top-left. Removed the legacy cruft. This was used to spawn mines as in Generals China ability before and recycled.